File: src/2971_find_polygon_with_the_largest_perimeter.py
class Solution:
    def largestPerimeter(self, nums: list[int]) -> int:
        nums.sort()
        curr = sum(nums)
        while nums and curr <= nums[-1] * 2:
            curr -= nums.pop()
        return sum(nums) if len(nums) > 2 else -1

File: src/2971_find_polygon_with_the_largest_perimeter.cpp
#include <algorithm>
#include <vector>

class Solution {
 public:
  auto largestPerimeter(std::vector<int>& nums) -> long long {
    std::sort(nums.begin(), nums.end());
    long long res = -1;
    long long cur = 0;
    for (int num : nums) {
      if ((cur += num) > num * 2) {
        res = cur;
      }
    }
    return res;
  }
};
